### ISSUE TITLE
Always propose Python 3 in the installation documentation

### ISSUE THREAD

**Issue by dmwyatt** [Kozea/WeasyPrint#405]
The instructions say 

> Debian 8.0 Jessie or newer, **Ubuntu 14.04 Trusty or newer**:
> 'sudo apt-get install python-dev python-pip python-lxml python-cffi libcairo2 libpango1.0-0 libgdk-pixbuf2.0-0 shared-mime-info'

On 16.04 I had to include `libffi-dev` or I received the following mountain of errors.  

```
Collecting weasyprint
  Using cached WeasyPrint-0.34-py3-none-any.whl
Collecting cffi>=0.6 (from weasyprint)
  Downloading cffi-1.9.1-cp35-cp35m-manylinux1_x86_64.whl (398kB)
    100% |████████████████████████████████| 399kB 3.1MB/s 
Collecting cairocffi>=0.5 (from weasyprint)
  Using cached cairocffi-0.7.2.tar.gz
    Complete output from command python setup.py egg_info:
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libffi' found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libffi' found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libffi' found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libffi' found
    Package libffi was not found in the pkg-config search path.
    Perhaps you should add the directory containing `libffi.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'libffi' found
    c/_cffi_backend.c:15:17: fatal error: ffi.h: No such file or directory
    compilation terminated.
    Traceback (most recent call last):
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/unixccompiler.py", line 118, in _compile
        extra_postargs)
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/ccompiler.py", line 909, in spawn
        spawn(cmd, dry_run=self.dry_run)
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/spawn.py", line 36, in spawn
        _spawn_posix(cmd, search_path, dry_run=dry_run)
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/spawn.py", line 159, in _spawn_posix
        % (cmd, exit_status))
    distutils.errors.DistutilsExecError: command 'gcc' failed with exit status 1
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/core.py", line 148, in setup
        dist.run_commands()
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/dist.py", line 955, in run_commands
        self.run_command(cmd)
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/command/bdist_egg.py", line 161, in run
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/command/bdist_egg.py", line 147, in call_command
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/command/install_lib.py", line 11, in run
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/command/install_lib.py", line 107, in build
        self.run_command('build_ext')
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/cmd.py", line 313, in run_command
        self.distribution.run_command(command)
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/dist.py", line 974, in run_command
        cmd_obj.run()
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/command/build_ext.py", line 77, in run
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/command/build_ext.py", line 338, in run
        self.build_extensions()
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/command/build_ext.py", line 447, in build_extensions
        self._build_extensions_serial()
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/command/build_ext.py", line 472, in _build_extensions_serial
        self.build_extension(ext)
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/command/build_ext.py", line 198, in build_extension
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/command/build_ext.py", line 532, in build_extension
        depends=ext.depends)
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/ccompiler.py", line 574, in compile
        self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/unixccompiler.py", line 120, in _compile
        raise CompileError(msg)
    distutils.errors.CompileError: command 'gcc' failed with exit status 1
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/sandbox.py", line 156, in save_modules
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/sandbox.py", line 197, in setup_context
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/sandbox.py", line 246, in run_setup
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/sandbox.py", line 276, in run
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/sandbox.py", line 245, in runner
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/sandbox.py", line 47, in _execfile
      File "/tmp/easy_install-sf8wkj1o/cffi-1.9.1/setup.py", line 202, in <module>
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/core.py", line 163, in setup
        raise SystemExit("error: " + str(msg))
    SystemExit: error: command 'gcc' failed with exit status 1
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/command/easy_install.py", line 1100, in run_setup
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/sandbox.py", line 249, in run_setup
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/contextlib.py", line 77, in __exit__
        self.gen.throw(type, value, traceback)
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/sandbox.py", line 197, in setup_context
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/contextlib.py", line 77, in __exit__
        self.gen.throw(type, value, traceback)
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/sandbox.py", line 168, in save_modules
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/sandbox.py", line 143, in resume
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/pkg_resources/_vendor/six.py", line 685, in reraise
        raise value.with_traceback(tb)
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/sandbox.py", line 156, in save_modules
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/sandbox.py", line 197, in setup_context
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/sandbox.py", line 246, in run_setup
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/sandbox.py", line 276, in run
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/sandbox.py", line 245, in runner
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/sandbox.py", line 47, in _execfile
      File "/tmp/easy_install-sf8wkj1o/cffi-1.9.1/setup.py", line 202, in <module>
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/core.py", line 163, in setup
        raise SystemExit("error: " + str(msg))
    SystemExit: error: command 'gcc' failed with exit status 1
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-ci4qher5/cairocffi/setup.py", line 65, in <module>
        **cffi_args
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/distutils/core.py", line 108, in setup
        _setup_distribution = dist = klass(attrs)
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/dist.py", line 315, in __init__
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/dist.py", line 361, in fetch_build_eggs
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/pkg_resources/__init__.py", line 851, in resolve
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/pkg_resources/__init__.py", line 1123, in best_match
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/pkg_resources/__init__.py", line 1135, in obtain
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/dist.py", line 428, in fetch_build_egg
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/command/easy_install.py", line 664, in easy_install
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/command/easy_install.py", line 694, in install_item
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/command/easy_install.py", line 875, in install_eggs
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/command/easy_install.py", line 1114, in build_and_install
      File "/home/dmwyatt/miniconda3/envs/msm/lib/python3.5/site-packages/setuptools-27.2.0-py3.5.egg/setuptools/command/easy_install.py", line 1102, in run_setup
    distutils.errors.DistutilsError: Setup script exited with error: command 'gcc' failed with exit status 1
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-ci4qher5/cairocffi/

```

**Comment by liZe** [comment]
This problem happens because Python3 became the default Python version installed on Ubuntu 16.04+ (and other distros as well). It means that we have to change the documentation to make users install `python3-*` packages instead of `python-*` for distributions having Python3 installed as default.

TL;DR: `apt-get install python3-dev python3-lxml python3-cffi` should work.

**Comment by liZe** [comment]
I think that we should only propose Python3 installation for Linux, as it's already done for MacOS and Windows.

(Before you ask, dear reader: yes, that's a first step to drop Python2 support soon.)
